src/eins-hwinfo.c

```diff
@@ -394,18 +459,33 @@ record_computer_hwinfo (gpointer unused)
     emtr_event_recorder_record_event (emtr_event_recorder_get_default (),
                                       COMPUTER_HWINFO_EVENT,
                                       g_steal_pointer (&payload));
+  set_next_record_time ();
+
+  /* The interval of first record after each boot usually is not 24 hour. */
+  if (record_first_time ())
+    {
+      eins_boottimeout_add_seconds (RECORD_COMPUTER_HWINFO_INTERVAL_SECONDS,
+                                    record_computer_hwinfo,
+                                    NULL);
```

Comment:
You don't actually need this helper function with static variable – you can pass data to the callback. Suggested changes:

```suggestion
  if (GPOINTER_TO_UINT (is_first_call))
    {
      eins_boottimeout_add_seconds (RECORD_COMPUTER_HWINFO_INTERVAL_SECONDS,
                                    record_computer_hwinfo,
                                    GUINT_TO_POINTER (FALSE));
```

and others at the relevant points.

Reply:
ok